### PR TITLE
Features/world bank okr

### PR DESCRIPTION
**Pull request by lpi-tn**
This pull request introduces comprehensive support for the World Bank Open Knowledge Repository (OKR) as a new data source. It adds a new collector, integrates the World Bank OKR data model into the processing pipeline, and provides extensive unit tests to ensure robust handling of OKR records, including edge cases and error scenarios. Additionally, it updates dependencies and makes minor improvements to existing test files.

**World Bank OKR Integration:**

* Added the `WorldBankOpenKnowledgeRepositoryCollector` class to handle the collection and parsing of OKR records, including logic for extracting URLs, DOIs, external IDs, and handling deleted records. (`welearn_datastack/collectors/world_bank_okr.py`)
* Integrated the `WorldBankOKRRecord` model into the `WrapperRawData` class, enabling downstream processing and error handling for OKR documents. (`welearn_datastack/data/db_wrapper.py`) [[1]](diffhunk://#diff-53eafbc3592df174464838e07efc42fa9ce28101c6572e20bc9b7c726dab39c9R9) [[2]](diffhunk://#diff-53eafbc3592df174464838e07efc42fa9ce28101c6572e20bc9b7c726dab39c9L41-R44)

**Testing Enhancements:**

* Added a comprehensive test suite for the World Bank OKR plugin, covering author extraction, license handling, content extraction (PDF and TXT), error scenarios, and the main processing pipeline. (`tests/document_collector_hub/plugins_test/test_world_bank_okr.py`)
* Introduced unit tests for the `WorldBankOKRRecord` data model to validate parsing and field extraction from example XML records. (`tests/source_models/test_world_bank_okr.py`)

**Configuration and Constants:**

* Updated the list of accepted Creative Commons licenses to include `by/3.0/igo/`, ensuring correct license validation for World Bank OKR content. (`welearn_datastack/constants.py`)
* Resolved a minor dependency version conflict for `python-dotenv` in the project configuration. (`pyproject.toml`)

**Minor Cleanups:**

* Removed unused imports and redundant assertions in existing test files to streamline the codebase. (`tests/document_collector_hub/plugins_test/test_unesdoc.py`) [[1]](diffhunk://#diff-d04680e4d0b70c56a711d9c46f18c2317fa8a50078fdbb16956ff3ab675709f1L6-R7) [[2]](diffhunk://#diff-d04680e4d0b70c56a711d9c46f18c2317fa8a50078fdbb16956ff3ab675709f1L116)
* Cleaned up commented code in enumerations for clarity. (`welearn_datastack/data/enumerations.py`)

### CODE REVIEW

File: welearn_datastack/collectors/world_bank_okr.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        except IndexError:
+            raise NotEnoughData("No identifier found in header of record")
+        return oai_pmh_id
+
+    def _is_deleted(self, xml_input: XMLData) -> bool:
+        full_record = XMLExtractor(xml_input.content)
+        ret = full_record.extract_content_attribute_filter(
+            tag="header", attribute_name="status", attribute_value="deleted"
+        )
+        return not bool(ret)
```
Comment (sandragjacinto) [inline welearn_datastack/collectors/world_bank_okr.py]: Je sais que cette ligne est correcte, donc c’est une suggestion pour que ce soit plus facile à lire et à comprendre la logique, car je trouve que ce que tu as utilisé est un peu contre-intuitif.

```suggestion
        return len(ret) == 0
```

File: welearn_datastack/collectors/world_bank_okr.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    def _extract_world_bank_okr_document(
+        self, world_bank_okr_api_response: XMLExtractor
+    ) -> list[WeLearnDocument]:
+        urls: List[WeLearnDocument] = []
+        records = world_bank_okr_api_response.extract_content(tag="record")
+        for record in records:
+            external_id = self._extract_external_id(record)
+            if not self._is_deleted(record):
```
Comment (sandragjacinto) [inline welearn_datastack/collectors/world_bank_okr.py]: lol la preuve que je suis confuse avec la methode is deleted 🤔  si ce n'est pas is_deleted alors on log que c'est deleted. je suis perdue 🦞

File: welearn_datastack/data/source_models/world_bank_okr.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    fileGrp: list[RecordFile]
+
+    @classmethod
+    def _extract_file_grp(cls, value: XMLExtractor) -> list[dict]:
+        ret = []
+        file_grp = value.extract_content(tag="fileGrp")[0].content
+        for f in XMLExtractor(file_grp).extract_content(tag="file"):
+            f_ret = {k.lower(): v for k, v in f.attributes.items()}
+            flocat_xml = XMLExtractor(f.content).extract_content(tag="FLocat")
+            print(0)
```
Comment (sandragjacinto) [inline welearn_datastack/data/source_models/world_bank_okr.py]:
```suggestion
```

**Review by sandragjacinto — APPROVED** [review]

File: welearn_datastack/plugins/rest_requesters/world_bank_okr.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        details = {
+            "authors": authors,
+            "topics": topics,
+            "publication_date": publication_date,
+            "doi": raw_data.identifiers.doi,
+        }
+
+        return details
+
+    def _extract_full_content(self, record: WorldBankOKRRecord) -> tuple[str, bool]:
```
Comment (jmsevin) [inline welearn_datastack/plugins/rest_requesters/world_bank_okr.py:130]: Don't you have some similar processing steps in some other corpus plugins? If you do, maybe some parts could be refactorized...

**Review by jmsevin — APPROVED** [review]

File: welearn_datastack/plugins/rest_requesters/world_bank_okr.py (hunk 4 above)
Comment (lpi-tn) [reply welearn_datastack/plugins/rest_requesters/world_bank_okr.py:130]: It's similar and I already try to factorize at maximum level, but this works is included in my big refacto project

File: welearn_datastack/collectors/world_bank_okr.py (hunk 2 above)
Comment (lpi-tn) [reply welearn_datastack/collectors/world_bank_okr.py]: J'ai fixé ça en rendant logique le retour de la fonction et en virant la double négation
